### PR TITLE
[xray-exporter] Take into account scenario statuses in after and before scenario steps

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: vividus-to-xray-exporter/src/main/java/org/vividus/xray/factory/TestExecutionFactory.java [hunk 1]
```diff
@@ -83,9 +82,11 @@ private static TestExecutionItem createTestInfo(Entry<String, Scenario> scenario
         return test;
     }
 
-    private static TestExecutionItemStatus calculateStatus(List<Step> steps)
+    private static TestExecutionItemStatus calculateStatus(AbstractStepsContainer stepsContainer)
     {
-        return steps.stream().allMatch(step -> "successful".equals(step.getOutcome())) ? TestExecutionItemStatus.PASS
-                : TestExecutionItemStatus.FAIL;
+        return Stream.of(stepsContainer.getSteps(), stepsContainer.getAfterScenarioSteps())
```
Comment (valfirst) [inline vividus-to-xray-exporter/src/main/java/org/vividus/xray/factory/TestExecutionFactory.java:87]: why is `BeforeScenarioSteps` not taken into account?

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-bdd-engine/src/main/java/org/vividus/bdd/model/jbehave/Scenario.java [hunk 2]
```diff
@@ -64,9 +58,9 @@ public void setExamples(Examples examples)
 
     public List<Step> collectSteps()
     {
-        if (steps != null)
+        if (getSteps() != null)
         {
-            return steps;
+            return getSteps();
         }
         return Optional.ofNullable(examples)
```
Comment (valfirst) [inline vividus-bdd-engine/src/main/java/org/vividus/bdd/model/jbehave/Scenario.java:65]: also I believe you should somehow consider before/after scenario steps of Examples

Comment (uarlouski) [reply vividus-bdd-engine/src/main/java/org/vividus/bdd/model/jbehave/Scenario.java:65]: this method is only needed to get steps that go into Xray scenario

Comment (uarlouski) [reply vividus-bdd-engine/src/main/java/org/vividus/bdd/model/jbehave/Scenario.java:65]: can rename it to collectScenarioSteps

**Review by valfirst — APPROVED** [review]

**Pull request merged**